server/engine/src/main/java/org/eclipse/lsp/cobol/dialects/ibm/TransformTreeStage.java

```diff
@@ -232,63 +234,81 @@ private Optional<CompilerDirectiveOption> getCompilerDirective(String compilerOp
 
     // ENRICHMENT
     ProcessingPhase e = ProcessingPhase.ENRICHMENT;
-    ctx.register(e, SectionNameNode.class, new SectionNameNodeEnricher(symbolAccumulatorService));
-    ctx.register(e, ParagraphNameNode.class, new ParagraphNameNodeEnricher(symbolAccumulatorService));
-    ctx.register(e, CodeBlockUsageNode.class, new CodeBlockUsageNodeEnricher(symbolAccumulatorService));
-    ctx.register(e, FunctionReference.class, new FunctionReferenceEnricher(symbolAccumulatorService));
-    ctx.register(e, ProgramIdNode.class, new ProgramIdEnricher(symbolAccumulatorService));
+    ctx.register(e, SectionNameNode.class, new SectionNameNodeEnricher(symbolAccumulator));
+    ctx.register(e, ParagraphNameNode.class, new ParagraphNameNodeEnricher(symbolAccumulator));
+    ctx.register(e, CodeBlockUsageNode.class, new CodeBlockUsageNodeEnricher(symbolAccumulator));
+    ctx.register(e, FunctionReference.class, new FunctionReferenceEnricher(symbolAccumulator));
+    ctx.register(e, ProgramIdNode.class, new ProgramIdEnricher(symbolAccumulator));
     ctx.register(e, ExitPerformNode.class, new ExitPerformEnricher());
 
     // Phase VALIDATION
     ProcessingPhase v = ProcessingPhase.VALIDATION;
-    ctx.register(v, VariableWithLevelNode.class, new VariableWithLevelCheck(CodeLayoutUtil.getProgramLayout(languageId, layoutStore)));
-    ctx.register(v, VariableWithLevelNode.class, new VariableNameCheck());
+    VariableWithLevelCheck variableWithLevelCheck = new VariableWithLevelCheck(CodeLayoutUtil.getProgramLayout(languageId, layoutStore));
+    ctx.register(v, ConditionDataNameNode.class, variableWithLevelCheck);
+    ctx.register(v, ElementaryItemNode.class, variableWithLevelCheck);
+    ctx.register(v, GroupItemNode.class, variableWithLevelCheck);
+    ctx.register(v, MultiTableDataNameNode.class, variableWithLevelCheck);
+    ctx.register(v, RenameItemNode.class, variableWithLevelCheck);
+    ctx.register(v, StandAloneDataItemNode.class, variableWithLevelCheck);
+    ctx.register(v, TableDataNameNode.class, variableWithLevelCheck);
+
+    VariableNameCheck variableNameCheck = new VariableNameCheck();
+    ctx.register(v, ConditionDataNameNode.class, variableNameCheck);
+    ctx.register(v, ElementaryItemNode.class, variableNameCheck);
+    ctx.register(v, GroupItemNode.class, variableNameCheck);
+    ctx.register(v, MultiTableDataNameNode.class, variableNameCheck);
+    ctx.register(v, RenameItemNode.class, variableNameCheck);
+    ctx.register(v, StandAloneDataItemNode.class, variableNameCheck);
+    ctx.register(v, TableDataNameNode.class, variableNameCheck);
+
     ctx.register(v, StatementNode.class, new StatementValidate());
+
     ctx.register(v, ElementaryNode.class, new ElementaryNodeCheck());
+
     ctx.register(v, GroupItemNode.class, new GroupItemCheck());
-    ctx.register(v, ObsoleteNode.class, new ObsoleteNodeCheck());
+    ctx.register(v, RemarksNode.class, new ObsoleteNodeCheck());
```

Comment:
This will start ignoring daco SortTableNode.

Reply:
it's ok, there is explicit registration for it in https://github.com/eclipse-che4z/che-che4z-lsp-for-cobol/blob/61d4e5c77dc11a16091f9e57c4c29f800fb0fca8/server/dialect-daco/src/main/java/org/eclipse/lsp/cobol/dialects/daco/DaCoDialect.java#L144